node-db/lesson1.md

```diff
@@ -9,8 +9,132 @@
 - This class is based on the [Node Girls Express Workshop](https://github.com/node-girls/express-workshop)
 ---
 
+# What is a server?
 
-# Intro
+Servers are computer programs that receive requests from other programs, the *clients* and send back a response e.g share data, information or hardware and software resources.
+
+### ...and what is a server in plain English?
+
+A server is a computer program. Its job is to send and receive data.
+
+Let's take a website for example.  A website is just a collection of HTML and CSS files, images, maybe some javascript files. When you type a website address in your browser's address bar, the browser (client) sends a **request** to the server that lives at that address. The browser asks the server to give it the files it needs to display the website properly.
+
+![Server flow](https://files.gitter.im/heron2014/FiiK/server.png)
+
+# What is Node.js?
+
+  > [Node.js®](https://nodejs.org/en/) is a **JavaScript runtime** built on Chrome's **V8 JavaScript engine**. Node.js uses an **event-driven**, **non-blocking** I/O model that makes it lightweight and efficient.
+  
+# What is it used for? 
+
+- web servers, so creating a dynamic websites
+- set up a local web development environment
+- easier to build desktop applications with Electron: Slack, Visual Code, Atom 
+- some of the biggest companies use Node.js in production: Netflix, Walmart, IBM, etc.
+- JavaScript everywhere (used to be PHP, Python, JavaScript, MySQL, Apache, now JavaScript full stack)
+
+# A simple Node.js server
+
+**No respond**  
+```javascript
+var http = require('http'); 	
+
+var server = http.createServer();
+
+server.listen(5000); 	
+
+console.log('Node.js web server at port 5000 is running..')
+```
+
+**Simple respond, no routing**
+```javascript
+var http = require('http'); 	
+
+var server = http.createServer(function (req, res) {
+	res.end("Hello World!")
+});
+
+server.listen(5000); 	
+
+console.log('Node.js web server at port 5000 is running..')
+```
+
+**Simple server with one route**    
+```javascript
+var http = require('http'); 	
+
+var server = http.createServer(function (req, res) {
+
+  //check the URL of the current request
+  if (req.url == '/') {
+  
+      console.log("Received a new request at " + Date())
+
+      // set response header
+      res.writeHead(200, { 'Content-Type': 'text/html' }); 
+
+      // set response content    
+      res.write('<html><body><p>This is home Page.</p></body></html>');
+      res.write('The time is: ' + Date());
+      res.end();
+    
+   } else {
+   
+      res.end('Invalid Request!');
+      
+   }
+});
+
+server.listen(5000); 	
+
+console.log('Node.js web server at port 5000 is running..')
+```
```

Comment:
it looks great .. could you just use const and let for the http example to be consistent with the rest .. also just have the two routes example (no need for single route imo).

Then maybe have an exercise for them to add a /mentor route